src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -344,102 +344,158 @@ public WorkspacePanel(final MCreator mcreator) {
+				PreferencesManager.PREFERENCES.hidden.workspaceModElementIconSize = PreferencesData.WorkspaceIconSize.TILES;
+				updateElementListRenderer();
+			}
+		});
+		tilesIcons.setSelected(PreferencesManager.PREFERENCES.hidden.workspaceModElementIconSize
+				== PreferencesData.WorkspaceIconSize.TILES);
+		Arrays.stream(tilesIcons.getChangeListeners()).forEach(e -> e.stateChanged(new ChangeEvent(tilesIcons)));
+		ComponentUtils.deriveFont(tilesIcons, 12);
+		tilesIcons.setBorder(BorderFactory.createEmptyBorder(3, 5, 3, 5));
+		modElementsBar.add(tilesIcons);
+
+		JRadioButtonMenuItem largeIcons = new JRadioButtonMenuItem(L10N.t("workspace.elements.list.large"));
 		largeIcons.setCursor(new Cursor(Cursor.HAND_CURSOR));
-		largeIcons.setIcon(UIRES.get("16px.large.gif"));
 		largeIcons.setContentAreaFilled(false);
 		largeIcons.setOpaque(false);
-		largeIcons.addChangeListener(e -> {
-			if (largeIcons.isSelected())
-				largeIcons.setForeground(Color.white);
-			else
-				largeIcons.setForeground(Color.darkGray.brighter());
-		});
 		largeIcons.addActionListener(e -> {
 			if (largeIcons.isSelected()) {
-				PreferencesManager.PREFERENCES.hidden.workspaceIconSize = PreferencesData.WorkspaceIconSize.LARGE;
+				PreferencesManager.PREFERENCES.hidden.workspaceModElementIconSize = PreferencesData.WorkspaceIconSize.LARGE;
 				updateElementListRenderer();
 			}
 		});
-		largeIcons.setSelected(
-				PreferencesManager.PREFERENCES.hidden.workspaceIconSize == PreferencesData.WorkspaceIconSize.LARGE);
+		largeIcons.setSelected(PreferencesManager.PREFERENCES.hidden.workspaceModElementIconSize
+				== PreferencesData.WorkspaceIconSize.LARGE);
 		Arrays.stream(largeIcons.getChangeListeners()).forEach(e -> e.stateChanged(new ChangeEvent(largeIcons)));
 		ComponentUtils.deriveFont(largeIcons, 12);
-		largeIcons.setBorder(BorderFactory.createEmptyBorder(0, 5, 0, 5));
+		largeIcons.setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
 		modElementsBar.add(largeIcons);
 
-		JToggleButton smallIcons = new JToggleButton(L10N.t("workspace.elements.list.small"));
+		JRadioButtonMenuItem mediumIcons = new JRadioButtonMenuItem(L10N.t("workspace.elements.list.medium"));
+		mediumIcons.setCursor(new Cursor(Cursor.HAND_CURSOR));
+		mediumIcons.setContentAreaFilled(false);
+		mediumIcons.setOpaque(false);
+		mediumIcons.addActionListener(e -> {
+			if (mediumIcons.isSelected()) {
+				PreferencesManager.PREFERENCES.hidden.workspaceModElementIconSize = PreferencesData.WorkspaceIconSize.MEDIUM;
+				updateElementListRenderer();
+			}
+		});
+		mediumIcons.setSelected(PreferencesManager.PREFERENCES.hidden.workspaceModElementIconSize
+				== PreferencesData.WorkspaceIconSize.MEDIUM);
+		Arrays.stream(mediumIcons.getChangeListeners()).forEach(e -> e.stateChanged(new ChangeEvent(mediumIcons)));
+		ComponentUtils.deriveFont(mediumIcons, 12);
+		mediumIcons.setBorder(BorderFactory.createEmptyBorder(3, 5, 3, 5));
+		modElementsBar.add(mediumIcons);

```

Comment:
Why are you adding menu items to top bar? Please re-read your changes and fix accordingly questionable parts..